helm/providers/charts/alarm/templates/pkgAlarmProvider.yaml

```diff
@@ -0,0 +1,79 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor
+# license agreements; and to You under the Apache License, Version 2.0.
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.alarmprovider.name | quote }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    name: alarmprovider
+spec:
+  replicas: {{ .Values.alarmprovider.replicaCount }}
+  template:
+    metadata:
+      labels:
+        name: {{ .Values.alarmprovider.name | quote }}
+    spec:
+      restartPolicy: {{ .Values.alarmprovider.restartPolicy }}
+      volumes:
+        - name: alarm-logs
+          persistentVolumeClaim:
+            claimName: {{ .Values.alarmprovider.persistence.pvcName | quote }}
+      containers:
+      - name: {{ .Values.alarmprovider.name | quote }}
+        imagePullPolicy: {{ .Values.alarmprovider.imagePullPolicy | quote }}
+        image: {{ .Values.alarmprovider.image | quote }}
+        ports:
+        - name: kafkaprovider
+          containerPort: {{ .Values.alarmprovider.apiPort }}
+        env:
+        - name: "DB_PROTOCOL"
+          valueFrom:
+            configMapKeyRef:
+              name: db.config
+              key: db_protocol
+        - name: "DB_HOST"
+          valueFrom:
+            configMapKeyRef:
+              name: db.config
+              key: db_host_port
+        - name: "DB_USERNAME"
+          valueFrom:
+            secretKeyRef:
+              name: db.auth
+              key: db_username
+        - name: "DB_PASSWORD"
+          valueFrom:
+            secretKeyRef:
+              name: db.auth
+              key: db_password
+        - name: "DB_PREFIX"
+          value: "{{- .Values.alarmprovider.dbPrefix -}}"
+        - name: "ROUTER_HOST"
+          valueFrom:
+            configMapKeyRef:
+              name: whisk.config
+              key: whisk_api_host_nameAndPort
+        - name: "ENDPOINT_AUTH"
+          valueFrom:
+            configMapKeyRef:
+              name: whisk.config
+              key: whisk_api_host_nameAndPort
+        volumeMounts:
+          - name: alarm-logs
+            mountPath: /logs
+
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
```

Comment:
Fixed.

Reply:
volumeClaimTemplates is not supported by a Deployment definition. So I cannot use volumeClaimTemplates here.

Reply:
There are probably two ways forward than, either use a StatefulSet instead of a Deployment so you can use volumeClaimTemplates or leave it as is and just a command in the Values.yaml that alarmProvider.numReplicas currently must have the value 1.   I think the StatefulSet is probably preferable, but either option would be ok.

Reply:
I will think about it. I think the key point is that whether we create multiple alarm providers or single. If multiple alarm providers are created, we may need more settings to support that.

Reply:
I think we support single alarm provider as for now. I will add a comment in Values.yaml.